src/container/srv_target.c

```diff
@@ -1513,7 +1542,12 @@ ds_cont_aggregate_ult(void *arg)
 
 		if (dss_xstream_exiting(dmi->dmi_xstream))
 			break;
-		ABT_thread_yield();
+
+		if (timeout > 0)
```

Comment:
check 'time / NSEC_PER_SEC > 0'? or check it and call ABT_thread_yield() on zero in dss_sched_timeout()?